src/main/kotlin/app/revanced/patches/all/activity/exportAll/patch/ExportAllActivitiesPatch.kt

```diff
@@ -0,0 +1,44 @@
+package app.revanced.patches.all.activity.exportAll.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.ResourceContext
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.ResourcePatch
+import app.revanced.patcher.patch.annotations.Patch
+
+@Patch(false)
+@Name("export-all-activities")
+@Description("Makes all app activities exportable.")
+@Version("0.0.1")
+class ExportAllActivitiesPatch : ResourcePatch {
+    override fun execute(context: ResourceContext): PatchResult {
+        context.xmlEditor["AndroidManifest.xml"].use { editor ->
+            val document = editor.file
+            val activities = document.getElementsByTagName("activity")
+
+            for(i in 0..activities.length) {
+                activities.item(i)?.let {
+                    val exportedAttribute = it.attributes.getNamedItem(EXPORTED_FLAG)
+                    if (exportedAttribute != null) {
+                        if(exportedAttribute.nodeValue != "true")
+                            exportedAttribute.nodeValue = "true"
+                        return@let
+                    }
+
+                    document.createAttribute(EXPORTED_FLAG)
+                        .apply { value = "true" }
+                        .let(it.attributes::setNamedItem)
```

Comment:
Why is this required in addition to the code above?

Reply:
Apps targeting a SDK below Android 12 are not required to include the attribute iirc so this code is for adding it if it doesn't already exist.